### PR TITLE
Ensure that pre-labelled nodes are not re-labelled

### PR DESCRIPTION
**Pull request by parthdhanjal**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1759051
Signed-off-by: parthdhanjal <dparth@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx [hunk 1]
```diff
@@ -194,14 +194,27 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({ data, loaded, ocsProp
...
     return selectedNodes.map((node: NodeKind) => {
-      const patch = [
-        {
-          op: 'add',
-          path: '/metadata/labels/cluster.ocs.openshift.io~1openshift-storage',
-          value: '',
-        },
-      ];
-      return k8sPatch(NodeModel, node, patch);
+      if (!node.metadata.labels['cluster.ocs.openshift.io/openshift-storage']) {
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: create a constant for the label.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx [hunk 2]
```diff
@@ -233,20 +246,27 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({ data, loaded, ocsProp
...
     promises.push(...makeLabelNodesRequest(selectedData));
+
     // intentionally keeping the taint logic as its required in 4.3 and will be handled with checkbox selection
     // promises.push(...makeTaintNodesRequest(selectedData));
 
     const ocsObj = _.cloneDeep(ocsRequestData);
     ocsObj.spec.storageDeviceSets[0].dataPVCTemplate.spec.storageClassName = storageClass;
-    promises.push(k8sCreate(OCSServiceModel, ocsObj));
 
+    // Nested Promises failing lint test as well
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: You also need to handle the scenario when all the selected nodes are already labeled, and promises array will be empty.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx (hunk 1 above)
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: create a getLabel selector in console/share folder.

Export https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/selectors/index.ts#L8

and then
`getLabels(node)[cephStorageLabel]`

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx [hunk 3]
```diff
@@ -245,20 +258,27 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({
...
+          history.push(
+            `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
+              ocsProps.clusterServiceVersion.metadata.name
+            }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
+          );
+        })
+        .catch((err) => {
+          setProgress(false);
+          setError(err.message);
+        });
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: Not required to be changed. if we are handling above via if condition. 

If promise obj is empty we need to block user to select new nodes.

Comment (cloudbehl) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: @gnehapk can you provide proper details on this one.

Comment (gnehapk) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: @cloudbehl We need to handle the scenario where if labeling node api call failed, it should not call k8sCreate. So, its required I feel. Thoughts?

**Pull request closed without merging**